File: python/fixtures/accepted/collections.py
# Some members here are named to stay distinct across the whole accepted corpus: the
# differential boundary tier builds every fixture into ONE unit, as a real project is built,
# and a duplicate name is refused by `Unit::add_function`. Renaming one back would break that
# build rather than any rule this fixture tests.
def first_and_count(xs: list[int]) -> int:
    first = xs[0]
    count = len(xs)
    return first + count


def lookup(d: dict[str, int], key: str) -> int:
    return d[key]


def build_list() -> list[int]:
    xs = [1, 2, 3]
    return xs


def mapping() -> dict[str, int]:
    return {"a": 1, "b": 2}


def unique() -> set[int]:
    return {1, 2, 2}


def pair() -> str:
    t = (1, "a")
    return t[1]


def nested(d: dict[str, list[int]], key: str) -> int:
    inner = d[key]
    return inner[0]


def from_end(xs: list[int]) -> int:
    return xs[-1]


def characters(s: str) -> int:
    return len(s)

File: frontends/typescript/fixtures/accepted/collections.ts
// Named for the Python counterparts they pair with; see the note in arithmetic.ts.
export function sumList(xs: Array<number>): number {
    let sum: number = 0;
    for (const x of xs) {
        sum = sum + x;
    }
    return sum;
}

export function first_and_count(xs: Array<number>): number {
    let first: number = xs[0];
    let count: number = xs.length;
    return first + count;
}

export function build_list(): Array<number> {
    let xs: Array<number> = [1, 2, 3];
    return xs;
}

export function characters(s: string): number {
    return s.length;
}
